graph/graph.go

```diff
@@ -415,6 +417,42 @@ func (graph *Graph) ByParent() (map[string][]*image.Image, error) {
 	return byParent, err
 }
 
+// If the images and layers are in pulling chain, retain them.
+// If not, they may be deleted by rmi with dangling condition.
+func (graph *Graph) Retain(layerIDs []string, imgID string) {
+	for _, layerID := range layerIDs {
+		if imgs, exists := graph.retained[layerID]; exists {
+			graph.retained[layerID] = append(imgs, imgID)
+		} else {
+			graph.retained[layerID] = []string{imgID}
+		}
+	}
+}
+
+// If the images and layers have been pulled, unretain them.
+func (graph *Graph) Unretain(layerIDs []string, imgID string) {
```

Comment:
Hi @stevvooe ,
I think yours and mine both can do what I want to.
But yours seems actually more reliable. I will change my code.
Thanks.